The Wazuh indexer service randomly fails after a restart

Testing the latest 5.0.0 (beta1) central components, an error appeared with the Wazuh Indexer service, which is detected as being in a failed state after restarting the service.

This issue is appearing on redhat-10-arm64 and amazon-2023-arm64, but other systems may also fail because of it. The error appears randomly, but whenever it does occur, it happens after a restart.

As example, a new execution failed in the following systems, this affects DEB and RPM packages for all architectures available

```
AssertionError: The wazuh-indexer service is not active after restart in the following hosts:[0m
redhat-9-amd64 (got: failed)[0m
redhat-9-arm64 (got: failed)[0m
amazon-2023-arm64 (got: failed)[0m
redhat-10-arm64 (got: failed)[0m
ubuntu-24-amd64 (got: failed)[0m
```

Apparently, the restart command succeded, but the service fails shortly after

```
× wazuh-indexer.service - wazuh-indexer
     Loaded: loaded (/usr/lib/systemd/system/wazuh-indexer.service; bad; preset: disabled)
     Active: failed (Result: exit-code) since Wed 2026-04-15 10:27:03 UTC; 19min ago
   Duration: 30.418s
 Invocation: 71eef82fc4a041b18e0824b7ba42c8e9
       Docs: https://documentation.wazuh.com
    Process: 14618 ExecStart=/usr/share/wazuh-indexer/bin/systemd-entrypoint -p ${PID_DIR}/wazuh-indexer.pid --quiet (code=exited, status=1/FAILURE)
   Main PID: 14618 (code=exited, status=1/FAILURE)
   Mem peak: 1.7G
        CPU: 1min 21.612s

Apr 15 10:27:02 aio-redhat-10-arm64 wazuh-indexer[14618]:         at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:159)
Apr 15 10:27:02 aio-redhat-10-arm64 wazuh-indexer[14618]:         at org.opensearch.common.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:110)
Apr 15 10:27:02 aio-redhat-10-arm64 wazuh-indexer[14618]:         at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
Apr 15 10:27:02 aio-redhat-10-arm64 wazuh-indexer[14618]:         at org.opensearch.cli.Command.main(Command.java:101)
Apr 15 10:27:02 aio-redhat-10-arm64 wazuh-indexer[14618]:         at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:125)
Apr 15 10:27:02 aio-redhat-10-arm64 wazuh-indexer[14618]:         at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:91)
Apr 15 10:27:02 aio-redhat-10-arm64 wazuh-indexer[14618]: For complete error details, refer to the log at /var/log/wazuh-indexer/wazuh-cluster.log
Apr 15 10:27:03 aio-redhat-10-arm64 systemd[1]: wazuh-indexer.service: Main process exited, code=exited, status=1/FAILURE
Apr 15 10:27:03 aio-redhat-10-arm64 systemd[1]: wazuh-indexer.service: Failed with result 'exit-code'.
Apr 15 10:27:03 aio-redhat-10-arm64 systemd[1]: wazuh-indexer.service: Consumed 1min 21.612s CPU time, 1.7G memory peak.
```

Checking the wazuh-cluster.log, the following output was found.

```
[2026-04-15T10:26:59,197][INFO ][o.o.i.MergeSchedulerConfig] [node-6] Updating maxThreadCount from [0] to [2] and maxMergeCount from [0] to [7] for index .cti-consumers.
[2026-04-15T10:26:59,197][INFO ][o.o.i.MergeSchedulerConfig] [node-6] Initialized index .cti-consumers with maxMergeCount=7, maxThreadCount=2, autoThrottleEnabled=true
[2026-04-15T10:26:59,248][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-6] Cancelling the migration process.
[2026-04-15T10:26:59,378][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-6] Cancelling the migration process.
[2026-04-15T10:26:59,450][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-6] Cancelling the migration process.
[2026-04-15T10:26:59,477][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-6] Cancelling the migration process.
[2026-04-15T10:26:59,581][INFO ][o.o.i.r.RecoverySourceHandler] [node-6] [.opensearch-sap-detectors-config][0][recover to node-7] finalizing recovery took [19.2ms]
[2026-04-15T10:26:59,615][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-6] Cancelling the migration process.
[2026-04-15T10:26:59,683][INFO ][c.w.c.c.c.s.AbstractConsumerService] [node-6] Synchronization completed for consumer [public-iocs-5]. Updated: false
[2026-04-15T10:26:59,703][INFO ][c.w.c.j.j.CatalogSyncJob ] [node-6] ConsumerIocService synchronized successfully.
[2026-04-15T10:26:59,706][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-6] Cancelling the migration process.
[2026-04-15T10:26:59,727][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-6] Cancelling the migration process.
[2026-04-15T10:26:59,808][INFO ][o.o.i.r.RecoverySourceHandler] [node-6] [.opensearch-sap-log-types-config][0][recover to node-7] finalizing recovery took [7ms]
[2026-04-15T10:26:59,841][INFO ][o.o.a.u.d.DestinationMigrationCoordinator] [node-6] Cancelling the migration process.
[2026-04-15T10:26:59,853][INFO ][c.w.c.c.c.s.AbstractConsumerService] [node-6] Performing update for consumer [public-vulnerabilities-5] from offset [340310] to [342286]
[2026-04-15T10:26:59,861][INFO ][c.w.c.c.c.s.UpdateServiceImpl] [node-6] Starting content update for consumer [public-vulnerabilities-5] from [340310] to [342286]
[2026-04-15T10:27:02,812][ERROR][o.o.b.Bootstrap          ] [node-6] Exception
org.opensearch.OpenSearchTimeoutException: java.util.concurrent.TimeoutException: Timeout waiting for task.
	at org.opensearch.common.util.concurrent.FutureUtils.get(FutureUtils.java:96)
	at org.opensearch.action.support.AdapterActionFuture.actionGet(AdapterActionFuture.java:79)
	at org.opensearch.action.support.AdapterActionFuture.actionGet(AdapterActionFuture.java:68)
	at com.wazuh.setup.index.IndexStateManagement.initialize(IndexStateManagement.java:189)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
	at com.wazuh.setup.SetupPlugin.onNodeStarted(SetupPlugin.java:200)
	at org.opensearch.node.Node.lambda$start$2(Node.java:1972)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
	at org.opensearch.node.Node.start(Node.java:1972)
	at org.opensearch.bootstrap.Bootstrap.start(Bootstrap.java:347)
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:421)
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:168)
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:159)
	at org.opensearch.common.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:110)
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
	at org.opensearch.cli.Command.main(Command.java:101)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:125)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:91)
Caused by: java.util.concurrent.TimeoutException: Timeout waiting for task.
	at org.opensearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:257)
	at org.opensearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:82)
	at org.opensearch.common.util.concurrent.FutureUtils.get(FutureUtils.java:94)
	... 17 more
[2026-04-15T10:27:02,814][ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [node-6] uncaught exception in thread [main]
org.opensearch.bootstrap.StartupException: OpenSearchTimeoutException[java.util.concurrent.TimeoutException: Timeout waiting for task.]; nested: TimeoutException[Timeout waiting for task.];
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:172)
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:159)
	at org.opensearch.common.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:110)
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138)
	at org.opensearch.cli.Command.main(Command.java:101)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:125)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:91)
Caused by: org.opensearch.OpenSearchTimeoutException: java.util.concurrent.TimeoutException: Timeout waiting for task.
	at org.opensearch.common.util.concurrent.FutureUtils.get(FutureUtils.java:96)
	at org.opensearch.action.support.AdapterActionFuture.actionGet(AdapterActionFuture.java:79)
	at org.opensearch.action.support.AdapterActionFuture.actionGet(AdapterActionFuture.java:68)
	at com.wazuh.setup.index.IndexStateManagement.initialize(IndexStateManagement.java:189)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
	at com.wazuh.setup.SetupPlugin.onNodeStarted(SetupPlugin.java:200)
	at org.opensearch.node.Node.lambda$start$2(Node.java:1972)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
	at org.opensearch.node.Node.start(Node.java:1972)
	at org.opensearch.bootstrap.Bootstrap.start(Bootstrap.java:347)
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:421)
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:168)
	... 6 more
Caused by: java.util.concurrent.TimeoutException: Timeout waiting for task.
	at org.opensearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:257)
	at org.opensearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:82)
	at org.opensearch.common.util.concurrent.FutureUtils.get(FutureUtils.java:94)
	... 17 more
[2026-04-15T10:27:02,817][INFO ][o.o.s.a.r.AuditMessageRouter] [node-6] Closing AuditMessageRouter
[2026-04-15T10:27:02,817][INFO ][o.o.s.a.s.SinkProvider   ] [node-6] Closing DebugSink
[2026-04-15T10:27:02,817][INFO ][o.o.n.Node               ] [node-6] stopping ...
[2026-04-15T10:27:02,826][ERROR][c.w.c.c.c.i.ContentIndex ] [node-6] Failed to index document [CVE-2026-3719]: NodeClosedException[node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}]
[2026-04-15T10:27:02,826][ERROR][c.w.c.c.c.s.UpdateServiceImpl] [node-6] Failed to apply offset [340311] (type=CREATE, resource=CVE-2026-3719): java.util.concurrent.ExecutionException: NodeClosedException[node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}]
java.io.IOException: java.util.concurrent.ExecutionException: NodeClosedException[node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}]
	at com.wazuh.contentmanager.cti.catalog.index.ContentIndex.create(ContentIndex.java:223)
	at com.wazuh.contentmanager.cti.catalog.service.UpdateServiceImpl.applyOffset(UpdateServiceImpl.java:196)
	at com.wazuh.contentmanager.cti.catalog.service.UpdateServiceImpl.update(UpdateServiceImpl.java:116)
	at com.wazuh.contentmanager.cti.catalog.service.AbstractConsumerService.syncConsumerServices(AbstractConsumerService.java:364)
	at com.wazuh.contentmanager.cti.catalog.service.AbstractConsumerService.synchronize(AbstractConsumerService.java:144)
	at com.wazuh.contentmanager.jobscheduler.jobs.CatalogSyncJob.performSynchronization(CatalogSyncJob.java:150)
	at com.wazuh.contentmanager.jobscheduler.jobs.CatalogSyncJob.lambda$trigger$0(CatalogSyncJob.java:134)
	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:918)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.util.concurrent.ExecutionException: NodeClosedException[node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}]
	at org.opensearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:286)
	at org.opensearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:260)
	at org.opensearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:82)
	at com.wazuh.contentmanager.cti.catalog.index.ContentIndex.create(ContentIndex.java:220)
	... 10 more
Caused by: org.opensearch.node.NodeClosedException: node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}
	at org.opensearch.action.support.replication.TransportReplicationAction$ReroutePhase$2.onClusterServiceClose(TransportReplicationAction.java:1209)
	at org.opensearch.cluster.ClusterStateObserver$ContextPreservingListener.onClusterServiceClose(ClusterStateObserver.java:387)
	at org.opensearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onClose(ClusterStateObserver.java:276)
	at org.opensearch.cluster.service.ClusterApplierService.doStop(ClusterApplierService.java:221)
	at org.opensearch.common.lifecycle.AbstractLifecycleComponent.stop(AbstractLifecycleComponent.java:97)
	at org.opensearch.cluster.service.ClusterService.doStop(ClusterService.java:158)
	at org.opensearch.common.lifecycle.AbstractLifecycleComponent.stop(AbstractLifecycleComponent.java:97)
	at org.opensearch.node.Node.stop(Node.java:2002)
	at org.opensearch.node.Node.close(Node.java:2036)
	at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:89)
	at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:131)
	at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:81)
	at org.opensearch.bootstrap.Bootstrap$4.run(Bootstrap.java:214)
[2026-04-15T10:27:02,828][INFO ][o.o.c.c.Coordinator      ] [node-6] cluster-manager node [{node-1}{0C5Tg1qyTRyuNLKTgGdNaw}{LUn77zusTpiDvfz9F3rciQ}{172.31.33.1}{172.31.33.1:9300}{dimr}{shard_indexing_pressure_enabled=true}] failed, restarting discovery
org.opensearch.transport.NodeDisconnectedException: [node-1][172.31.33.1:9300][disconnected] disconnected
[2026-04-15T10:27:02,828][ERROR][c.w.c.c.c.s.UpdateServiceImpl] [node-6] Error during content update: java.util.concurrent.ExecutionException: NodeClosedException[node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}]
java.io.IOException: java.util.concurrent.ExecutionException: NodeClosedException[node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}]
	at com.wazuh.contentmanager.cti.catalog.index.ContentIndex.create(ContentIndex.java:223)
	at com.wazuh.contentmanager.cti.catalog.service.UpdateServiceImpl.applyOffset(UpdateServiceImpl.java:196)
	at com.wazuh.contentmanager.cti.catalog.service.UpdateServiceImpl.update(UpdateServiceImpl.java:116)
	at com.wazuh.contentmanager.cti.catalog.service.AbstractConsumerService.syncConsumerServices(AbstractConsumerService.java:364)
	at com.wazuh.contentmanager.cti.catalog.service.AbstractConsumerService.synchronize(AbstractConsumerService.java:144)
	at com.wazuh.contentmanager.jobscheduler.jobs.CatalogSyncJob.performSynchronization(CatalogSyncJob.java:150)
	at com.wazuh.contentmanager.jobscheduler.jobs.CatalogSyncJob.lambda$trigger$0(CatalogSyncJob.java:134)
	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:918)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.util.concurrent.ExecutionException: NodeClosedException[node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}]
	at org.opensearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:286)
	at org.opensearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:260)
	at org.opensearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:82)
	at com.wazuh.contentmanager.cti.catalog.index.ContentIndex.create(ContentIndex.java:220)
	... 10 more
Caused by: org.opensearch.node.NodeClosedException: node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}
	at org.opensearch.action.support.replication.TransportReplicationAction$ReroutePhase$2.onClusterServiceClose(TransportReplicationAction.java:1209)
	at org.opensearch.cluster.ClusterStateObserver$ContextPreservingListener.onClusterServiceClose(ClusterStateObserver.java:387)
	at org.opensearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onClose(ClusterStateObserver.java:276)
	at org.opensearch.cluster.service.ClusterApplierService.doStop(ClusterApplierService.java:221)
	at org.opensearch.common.lifecycle.AbstractLifecycleComponent.stop(AbstractLifecycleComponent.java:97)
	at org.opensearch.cluster.service.ClusterService.doStop(ClusterService.java:158)
	at org.opensearch.common.lifecycle.AbstractLifecycleComponent.stop(AbstractLifecycleComponent.java:97)
	at org.opensearch.node.Node.stop(Node.java:2002)
	at org.opensearch.node.Node.close(Node.java:2036)
	at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:89)
	at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:131)
	at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:81)
	at org.opensearch.bootstrap.Bootstrap$4.run(Bootstrap.java:214)
[2026-04-15T10:27:02,829][INFO ][c.w.c.c.c.s.UpdateServiceImpl] [node-6] Resetting consumer [public-vulnerabilities-5] offset to 0 due to update failure.
[2026-04-15T10:27:02,830][ERROR][c.w.c.c.c.s.UpdateServiceImpl] [node-6] Failed to reset consumer: node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}
[2026-04-15T10:27:02,835][ERROR][c.w.c.ContentManagerPlugin] [node-6] Failed to schedule Telemetry Ping Job: node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}
[2026-04-15T10:27:02,887][INFO ][c.w.c.c.c.s.AbstractConsumerService] [node-6] Synchronization completed for consumer [public-vulnerabilities-5]. Updated: true
[2026-04-15T10:27:02,892][WARN ][c.w.c.c.c.s.AbstractConsumerService] [node-6] Failed to set consumer [public-vulnerabilities-5] status to [idle]: node closed {node-6}{V7fLTOQFSV-TbDEz5hf1Rw}{qz14T8ZdTtu55Guz-hI-rA}{172.31.80.147}{172.31.80.147:9300}{dimr}{shard_indexing_pressure_enabled=true}
[2026-04-15T10:27:02,893][INFO ][c.w.c.j.j.CatalogSyncJob ] [node-6] ConsumerCveService synchronized successfully.
[2026-04-15T10:27:02,898][INFO ][o.o.n.Node               ] [node-6] stopped
[2026-04-15T10:27:02,898][INFO ][o.o.n.Node               ] [node-6] closing ...
[2026-04-15T10:27:02,904][INFO ][o.o.s.a.i.AuditLogImpl   ] [node-6] Closing AuditLogImpl
[2026-04-15T10:27:02,907][INFO ][o.o.n.Node               ] [node-6] closed
```

This ZIP file contains all log files from /var/log/wazuh-indexer [indexer_log.zip](https://github.com/user-attachments/files/26746740/indexer_log.zip)


> [!NOTE]
> If this is expected and the tests must change, please notify us

All those systems use the following hardware

- AWS EC2: `c5ad.xlarge`
- vCPU: 4
- RAM: 8 GiB
- Disk: 30 GiB

--- a/plugins/setup/src/main/java/com/wazuh/setup/SetupPlugin.java
+++ b/plugins/setup/src/main/java/com/wazuh/setup/SetupPlugin.java
@@ -176,31 +176,35 @@ public class SetupPlugin extends Plugin implements ClusterPlugin, ActionPlugin {
     public void onNodeStarted(DiscoveryNode localNode) {
         // Initialize the indices only if this node is the cluster manager node.
         if (localNode.isClusterManagerNode()) {
+            this.threadPool
+                    .generic()
+                    .execute(
+                            () -> {
+                                // Apply cluster.default_number_of_replicas from opensearch.yml settings if present
+                                try {
+                                    String defaultNumberOfReplicas =
+                                            this.clusterService.getSettings().get(CLUSTER_DEFAULT_NUMBER_OF_REPLICAS);
+                                    if (defaultNumberOfReplicas != null) {
+                                        ClusterUpdateSettingsRequest request = new ClusterUpdateSettingsRequest();
+                                        request.persistentSettings(
+                                                Settings.builder()
+                                                        .put(CLUSTER_DEFAULT_NUMBER_OF_REPLICAS, defaultNumberOfReplicas)
+                                                        .build());
+                                        this.client
+                                                .admin()
+                                                .cluster()
+                                                .updateSettings(request)
+                                                .actionGet(PluginSettings.getTimeout(this.clusterService.getSettings()));
+                                        log.info(
+                                                "Successfully updated cluster.default_number_of_replicas to {}",
+                                                defaultNumberOfReplicas);
+                                    }
+                                } catch (Exception e) {
+                                    log.error("Failed to update cluster.default_number_of_replicas", e);
+                                }
 
-            // Apply cluster.default_number_of_replicas from opensearch.yml settings if present
-            try {
-                String defaultNumberOfReplicas =
-                        this.clusterService.getSettings().get(CLUSTER_DEFAULT_NUMBER_OF_REPLICAS);
-                if (defaultNumberOfReplicas != null) {
-                    ClusterUpdateSettingsRequest request = new ClusterUpdateSettingsRequest();
-                    request.persistentSettings(
-                            Settings.builder()
-                                    .put(CLUSTER_DEFAULT_NUMBER_OF_REPLICAS, defaultNumberOfReplicas)
-                                    .build());
-                    this.client
-                            .admin()
-                            .cluster()
-                            .updateSettings(request)
-                            .actionGet(PluginSettings.getTimeout(this.clusterService.getSettings()));
-                    log.info(
-                            "Successfully updated cluster.default_number_of_replicas to {}",
-                            defaultNumberOfReplicas);
-                }
-            } catch (Exception e) {
-                log.error("Failed to update cluster.default_number_of_replicas", e);
-            }
-
-            this.indices.forEach(Index::initialize);
+                                this.indices.forEach(Index::initialize);
+                            });
         }
     }
 

